ONSdigital/eq-survey-runner
# Update builds and instructions to align with AWS ElasticBeanstalk

### What is the context of this PR?
https://trello.com/c/74T6bv1m

This PR updates our builds (local/Docker) to align more closely with Amazon Elastic Beanstalk to match our deployment model. 

I have added a .python-version file with the version set to 3.4.2. In order to easily install the correct/exact version of Python I found pyenv useful (it is also what generates the .python-version file) so have updated the README to refer to this tool (https://github.com/yyuu/pyenv).

I also updated the Docker setup to be based on the Amazon AMI base image, install Apache and Python 3.4.2 and run the app behind mod_wsgi. It's not identical to AWS EB, but much closer than we had before.

The Dockerfile is now based on a base image we've added via https://github.com/ONSdigital/docker-aws-apache-wsgi/tree/python-3.4.2 which is auto-built by https://hub.docker.com/ as part of the onsdigital organisation (https://hub.docker.com/r/onsdigital/docker-aws-apache-wsgi/) this adds the basic AWS image, Apache, WSGI and Node/Yarn dependencies.

In addition, the Docker container has been updated to use some of the newer environment variables (e.g. developer logging) and removed the mounted volume as this stops the Docker image from re-building correctly when the file system changes. This change should make Docker re-run yarn compile if the filesystem changes (i.e. with `docker-compose build`) without having to re-build the whole container.

Note I removed the S3 instructions/environment vars from the README/config as it is no longer applicable since S3 support was removed a while ago.

Note I successfully got PyCharm to use the pyenv virtualenv by adding the virtualenv binary to the interpreters setup.

### How to review 
- Try using pyenv to install the correct Python version and setting a virtualenv by following the README
- Try launching via Docker to verify it works correctly
